crates/proto/src/xfer/dns_response.rs

```diff
@@ -88,9 +144,10 @@ type TimeoutFuture = Pin<
 // TODO: see https://github.com/bluejekyll/trust-dns/issues/383 for removing vec of messages and instead returning a Stream
 /// A DNS response object
 ///
-/// For Most DNS requests, only one response is expected, the exception is a multicast request.
+/// For Most DNS requests, only one response is expected, the exception are multicast requests and
+/// zone transferts.
 #[derive(Clone, Debug)]
-pub struct DnsResponse(Message);
+pub struct DnsResponse(SmallVec<[Message; 1]>);
```

Comment:
Sorry for the delay on merging this PR. I'm just trying to figure out how I feel about reintroducing a Vec of Message inside DnsResponse. I know it's the most expedient way to fix this, but I guess I was hoping that we might be able to move to a fully-streaming interface instead. This is ultimately my biggest issue with the change personally, but I don't know when we're going to be able to fix all the guts of the libraries to implement a real streaming interface here...

Thoughts?

Reply:
It doesn't seem reasonable to ask a contributor to overhaul stuff to become fully streaming for this to work, so we should probably just merge this and then fix it up when we make things streaming. This seems conceptually more correct even if the current implementation is a bit awkward, right?

Reply:
I think that's a convincing argument.